tiledb/sm/filesystem/s3.cc

```diff
@@ -1173,15 +1190,11 @@ Status S3::initiate_multipart_request(Aws::Http::URI aws_uri) {
         path_c_str + outcome_error_message(multipart_upload_outcome)));
   }
 
-  MultiPartUploadState state(
-      1,
-      Aws::String(aws_uri.GetAuthority()),
-      Aws::String(path),
-      Aws::String(multipart_upload_outcome.GetResult().GetUploadId()),
-      std::map<int, Aws::S3::Model::CompletedPart>());
-
-  assert(multipart_upload_states_.count(path_c_str) == 0);
-  multipart_upload_states_.emplace(std::move(path_c_str), std::move(state));
+  state->part_number = 1;
```

Comment:
maybe a vestige from an earlier commit, i don't see any reason to set each field individually instead of using the constructor